lib/backends/backend-wasm.ts

```diff
@@ -60,3 +53,32 @@ export class WasmBackend implements Backend, WasmOptions {
     }
   }
 }
+
+function defaultNumWorkers(): number {
+  if (typeof navigator !== 'undefined' && navigator) {
+    // by default use ([navigator.hardwareConcurrency / 2] - 1) workers
+    if (typeof navigator.hardwareConcurrency === 'number') {
+      return Math.max(Math.ceil(navigator.hardwareConcurrency / 2) - 1, 0);
```

Comment:
If navigator.hardwareConcurrency doesn't exist, is the default number of workers 0 or 1? I mean, what's the basis that two behaviors exists here? Is it possible to make the logic simple to return 0 or 1 without checking Safari case here?

Reply:
Safari browser ( on Apple devices ) does not expose API `navigator.hardwareConcurrency`, but it has the availability to launch web workers. use `1` as number of web worker helps to leverage the multithread capability and it's not too aggressive.

If it is not Safari browser, this should not happen to any modern browser. This is most likely because the browser is too old to support web worker.